### ISSUE TITLE
Format code with prettier on save

### ISSUE THREAD

**Issue by PullJosh** [PullJosh/leopard-website#59]
I think this should be a very strong default (possibly even the only option for now). Particularly for beginners, opinionated code formatting helps a lot.

**Comment by towerofnix** [comment]
Might be a little tricky to make sure the cursor sticks effectively where it was - so it doesn't shift somewhere a bit offset when you save - but this is presumably something other IDEs have already figured out, and maybe something prettier itself supports?

**Comment by PullJosh** [comment]
Yes indeed! https://prettier.io/docs/en/api.html#prettierformatwithcursorsource--options

> `formatWithCursor` both formats the code, and translates a cursor position from unformatted code to formatted code. This is useful for editor integrations, to prevent the cursor from moving when code is formatted.
